### PR TITLE
5922 Use sidebar error boundary and add error report

### LINKED ISSUE #5922: Add `ViewError` and `Click` events to Sidebar
This feature is a result of https://github.com/pixiebrix/pixiebrix-app/issues/3534

Add the following Mixpanel events to the Sidebar:
- `ViewError` that is reported when an error _view_ is shown to a user (e.g. ErrorBoundry, error toasts, alerts, etc.)
- `Click` events with properties that describe where & what the user clicked on, adding these especially to Activation configuration options & buttons

### PR DESCRIPTION
**Pull request by turbochef**
## What does this PR do?

- Closes #5922
- Add a `reportError` call to the sidebar `ErrorBoundary` and then swap the tabs to use the sidebar-specific error boundary rather than the generic one.
- Add `onError()` prop to the sidebar error boundary
- Add `reportEvent` to the tabs onError with context
- Add `reportEvent` to the submit and cancel buttons in the activation inputs

## Demo

before:
image

after:
image

## Checklist

- [ ] Add tests
- [x] Designate a primary reviewer: @mnholtz

### CODE REVIEW

File: src/sidebar/ErrorBoundary.tsx [hunk 1]
```diff
@@ -38,6 +39,7 @@ class ErrorBoundary extends Component<UnknownObject, State> {
   }
 
   static getDerivedStateFromError(error: Error) {
+    reportError(error);
```
Comment (turbochef) [inline src/sidebar/ErrorBoundary.tsx:42]: The ticket calls for a `reportEvent("ViewError")` but I assume `reportError` also sends mixpanel events. If needed I can add both.

File: src/sidebar/Tabs.tsx [hunk 2]
```diff
@@ -240,7 +271,13 @@ const Tabs: React.FC = () => {
               key={recipeToActivate.recipeId}
               eventKey={eventKeyForEntry(recipeToActivate)}
             >
-              <ErrorBoundary>
+              <ErrorBoundary
+                onError={() => {
+                  reportEvent("ViewError", {
+                    recipeToActivate: recipeToActivate.recipeId,
```
Comment (twschiller) [inline src/sidebar/Tabs.tsx:277]: I think this should have a panelType of "activate" or similar and then pass the recipeId for blueprintId?

Comment (turbochef) [reply src/sidebar/Tabs.tsx:277]: Makes sense. I'll add it.

File: src/sidebar/Tabs.tsx [hunk 3]
```diff
@@ -68,19 +68,28 @@ const TemporaryPanelTabPane: React.FC<{
...
 
   return (
     <Tab.Pane
       className={cx("full-height flex-grow", styles.paneOverrides)}
       eventKey={eventKeyForEntry(panel)}
     >
-      <ErrorBoundary>
+      <ErrorBoundary
+        onError={() => {
+          reportEvent("ViewError", {
```
Comment (grahamlangford) [inline src/sidebar/Tabs.tsx:80]: This is very vague, probably not that useful for product. Perhaps it should be "SidebarViewError"?

**Review by grahamlangford — APPROVED** [review]

**Review by mnholtz — APPROVED** [review]
Thanks @turbochef !

Would you add definitions to these events in the Mixpanel lexicon once we release?

**Pull request merged**